kyo-combinators/shared/src/main/scala/kyo/Combinators.scala

```diff
@@ -536,29 +473,43 @@ class CatchSomeAbortPartialOps[A, S, E, E1 <: E](effect: A < (Abort[E] & S)) ext
                     case e1: Result.Error[?]                   => Abort.get(e1)
                     case Result.Success(a)                     => a
             }
-    end apply
-end CatchSomeAbortPartialOps
 
-class SwapSomeAbortOps[A, S, E, E1 <: E](effect: A < (Abort[E] & S)) extends AnyVal:
+    def toChoice[ER](
+        using
+        ev: E => E1 | ER,
+        ct: SafeClassTag[E1],
+        reduce: Reducible[Abort[ER]],
+        flat: Flat[A],
+        frame: Frame
+    ): A < (S & reduce.SReduced & Choice) =
+        Abort.run[E1](effect.asInstanceOf[A < (Abort[E1 | ER] & S)]).map(e => Choice.get(e.fold(_ => Nil)(List(_))))
```

Comment:
Can we avoid the cast with the evidence?

Reply:
I don't see how. `ev` can convert `E` to `E1 | ER`, another type, of which the `E` is simply a parameter. @fwbrasil any thoughts here?

Reply:
Ah you're right. I think the cast is fine.

Reply:
yep, I think it's not possible to use the `E => E1 | ER ` evidence to transform the pending set